src/devices/bus/coco/cococart.cpp

```diff
@@ -678,10 +677,7 @@ device_cococart_interface::~device_cococart_interface()
 
 void device_cococart_interface::interface_config_complete()
 {
-	m_owning_slot = dynamic_cast<cococart_slot_device *>(device().owner());
-	m_host = m_owning_slot
-			? dynamic_cast<device_cococart_host_interface *>(m_owning_slot->owner())
-			: nullptr;
+	m_host = dynamic_cast<device_cococart_host_interface *>(m_owning_slot->owner());
 }
```

Comment:
You need to be prepared for this to run from `-validate`, `-listxml`, etc. in which case the device will be instantiated alone without the slot it expects.  In these cases, `m_owning_slot` will be null, and this read from the guard page at zero if you remove the check.